ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_math.py

```diff
@@ -1416,6 +1417,45 @@ def test_paddle_frac(
     )
 
 
+# diagonal
+@handle_frontend_test(
+    fn_tree="paddle.tensor.math.diagonal",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        min_num_dims=2,
+        max_num_dims=2,
+        min_dim_size=1,
+        max_dim_size=50,
+    ),
+    offset=helpers.ints(min_value=-10, max_value=50),
+    axes=st.lists(
+        helpers.ints(min_value=-2, max_value=1), min_size=2, max_size=2, unique=True
```

Comment:
similarly, the `axis` parameter can be any integer in the range of `[-rank, rank-1]`; where `rank=ndim(x)`. You'll need to modify the list generation here to generate axes conforming to the abovementioned range.